bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/handlers/FileOpenListener.java

```diff
@@ -38,6 +41,24 @@ public void partActivated(IWorkbenchPartReference partRef)
 			IProject project = file.getProject();
 			if (project == null)
 				return;
+			
+			try
+			{
+				IContentTypeManager contentTypeManager = Platform.getContentTypeManager();
+				IContentType contentType = contentTypeManager.findContentTypeFor(file.getFullPath().toOSString());
+				if (contentType != null)
+				{
+					String id = contentType.getId();
+					if (!(id.startsWith("org.eclipse.cdt.core.c") && (id.endsWith("Source") || id.endsWith("Header")))) {
+	                    return; // Skip files that are not C source/header files
+	                }
+				}
+			}
+			catch (Exception e)
+			{
+				Logger.log(e);
+			}
```

Comment:
LGTM!

Nit: Consider extracting this logic into a private method for better readability. You could then add a single condition to skip processing, like this: 
```
			if (project == null || !isCSourceOrHeader(file))
				return;
```
```
	private boolean isCSourceOrHeader(IFile file)
	{
		try
		{
			IContentTypeManager contentTypeManager = Platform.getContentTypeManager();
			IContentType contentType = contentTypeManager.findContentTypeFor(file.getFullPath().toOSString());
			if (contentType != null)
			{
				String id = contentType.getId();
				return id.startsWith("org.eclipse.cdt.core.c") && (id.endsWith("Source") || id.endsWith("Header")); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
			}
		}
		catch (Exception e)
		{
			Logger.log(e);
		}
		return false;
	}
```